lib/queue.js

```diff
@@ -26,16 +28,9 @@ module.exports = FetchQueue;
 
 FetchQueue.prototype = [];
 FetchQueue.prototype.add = function(protocol, domain, port, path, depth, callback) {
-
-    // For legacy reasons
-    if (depth instanceof Function) {
-        callback = depth;
-        depth = 1;
-    }
-
     depth = depth || 1;
-    callback = callback && callback instanceof Function ? callback : function() {};
-    var self = this;
+    callback = callback instanceof Function ? callback : noop;
```

Comment:
I only made this change due to the previous syntax screwing up syntax highlighting for me :grin: